tianshou/highlevel/params/dist_fn.py

```diff
@@ -20,32 +20,37 @@ def create_dist_fn(
 
 
 class DistributionFunctionFactoryCategorical(DistributionFunctionFactory):
+    def __init__(self, is_probs_input: bool = True):
+        """
+        :param is_probs_input: If True, the distribution function shall create a categorical distribution from a
+            tensor containing probabilities; otherwise the tensor is assumed to contain logits.
+        """
+        self.is_probs_input = is_probs_input
+
     def create_dist_fn(self, envs: Environments) -> TDistFnDiscrete:
         envs.get_type().assert_discrete(self)
-        return self._dist_fn
+        if self.is_probs_input:
+            return self._dist_fn_probs
+        else:
+            return self._dist_fn
 
+    # NOTE: Do not move/rename because a reference to the function can appear in persisted policies
     @staticmethod
-    def _dist_fn(p: torch.Tensor) -> torch.distributions.Categorical:
-        return torch.distributions.Categorical(logits=p)
+    def _dist_fn(logits: torch.Tensor) -> torch.distributions.Categorical:
+        return torch.distributions.Categorical(logits=logits)
+
+    # NOTE: Do not move/rename because a reference to the function can appear in persisted policies
+    def _dist_fn_probs(self, probs: torch.Tensor) -> torch.distributions.Categorical:
```

Comment:
make static? Or, on the contrary, make `_dist_fn` not static and remove this, in favor of `_dist_fn` taking care of choosing logits or probs.

Reply:
Indeed, the method should have been static; I will fix this.

> Or, on the contrary, make _dist_fn not static and remove this, in favor of _dist_fn taking care of choosing logits or probs.

This would be bad idea. In fact, it would constitute a bug. As I remarked in the comments, references to these functions appear in persisted policies. If you change the unbound method to a bound method, all persisted instances will cease to function.